akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -160,10 +197,44 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
       // Increment counter of events received since last NewDMGeneratingVersionEvent
       recoveredEventsCount_sinceLast_dmGeneratingVersion = recoveredEventsCount_sinceLast_dmGeneratingVersion + 1
       onReceiveRecover(event)
+    case offer: SnapshotOffer => {
+      if(!offer.isInstanceOf[FullSnapshotState]) {
+        throw new RuntimeException(s"Snapshot is not of the expected type ${FullSnapshotState.getClass.getName}) but was ${offer.getClass.getName}" )
+      }
+      val offerState = offer.snapshot.asInstanceOf[FullSnapshotState]
+      recoverStateFromSnapshot(offerState.localState)
+      onSnapshotRecovery(SnapshotOffer(offer.metadata, offerState.userData))
+    }
     case x:Any =>
       log.error(s"Ignoring msg of unknown type: ${x.getClass}")
   }
 
+
+  override def saveSnapshot(snapshot: Any): Unit = {
+    val actorState = StoredEnhancedPersistentActorState(
+      isProcessingEvent = isProcessingEvent,
+      pendingDurableMessage = pendingDurableMessage,
+        eventLogLevelInfo = eventLogLevelInfo,
+        recoveringEventLogLevelInfo = recoveringEventLogLevelInfo,
+        cmdLogLevelInfo = recoveringEventLogLevelInfo,
+        currentLogLevelInfo = currentLogLevelInfo,
+        prevLogLevelTryCommand = prevLogLevelTryCommand,
+        persistAndApplyEventHasBeenCalled = persistAndApplyEventHasBeenCalled,
+        dmGeneratingVersionFixedDeliveryIds = dmGeneratingVersionFixedDeliveryIds,
+        currentDmGeneratingVersion = currentDmGeneratingVersion,
+        addDmGeneratingVersionIfSavingEvents = addDmGeneratingVersionIfSavingEvents,
+        recoveredEventsCount_sinceLast_dmGeneratingVersion = recoveredEventsCount_sinceLast_dmGeneratingVersion,
+        processedDMs = processedDMs
+    )
+    super.saveSnapshot(FullSnapshotState(snapshot,actorState))
+  }
+
+
+  protected def onSnapshotRecovery(offer : SnapshotOffer): Unit = {
+
+    log.warning(s"Ignoring recovery from snapshot $offer, override this method to process offer")

```

Comment:
This will not work...

When you are given a snapshot, you have to accept it it or fail... If not, akka thinks you got it, and will only send you events newer than the snapshot

Reply:
Oh, ok so exception there.